sql/testdata/snapshot_unrelated_update

```diff
@@ -0,0 +1,93 @@
+statement ok
+CREATE TABLE t (id INT PRIMARY KEY, v string)
+
+statement ok
+GRANT ALL ON t TO testuser
+
+statement ok
+INSERT INTO t VALUES (1, 'A'), (2, 'B')
+
+# Test that two transactions both scanning the table can
+# perform non-conflicting updates without a restart.
+
+# First case: scan, scan, write, commit, write, commit.
+
+user testuser
+
+statement ok
+BEGIN TRANSACTION ISOLATION LEVEL SNAPSHOT
+
+query IT
+SELECT * FROM t
+----
+1 A
+2 B
+
+user root
+
+statement ok
+BEGIN TRANSACTION ISOLATION LEVEL SNAPSHOT
```

Comment:
Does this actually create another transaction? I thought you would see errTransactionInProgress